battle-engine.js

```diff
@@ -418,14 +410,14 @@ BattlePokemon = (() => {
 			}
 		}
 		return targets;
-	};
-	BattlePokemon.prototype.ignoringAbility = function () {
+	}
+	get ignoringAbility() {
```

Comment:
Also don't use getters for things that are clearly calculated from something else.

Getters are good for if you want to hide that a variable is actually running a function. Most of the time, you don't want to hide that because the fact that it's actually running a function is useful information to maintainers.

For instance, if I see `pokemon.ignoringAbility`, I'm going to think "oh shit, I'm going to need to change this variable when I start Gastro Acid. If I see `pokemon.ignoringAbility()`, I'm going to think "okay good this automatically figures out whether or not Gastro Acid is in effect.